src/vos/vos_common.c

```diff
@@ -502,6 +502,10 @@ vos_mod_init(void)
 	D_INFO("Set aggregate NVMe record threshold to %u blocks (blk_sz:%lu).\n",
 	       vos_agg_nvme_thresh, VOS_BLK_SZ);
 
+	d_getenv_bool("DAOS_DKEY_PUNCH_PROPAGATE", &vos_dkey_punch_propagate);
```

Comment:
i always forget.. if user does not set this env, d_getenv_bool  defaults to false?

Reply:
It does nothing if it's not set.  So the default in this case is false because I set it to false (or it gets zero initialized as a static)